### DISCUSSION TITLE
Guidelines for mimetype icons for packages, archives and more...

### DISCUSSION THREAD

**Discussion by madsrh** [ubuntu/yaru#3474] [General]
**Hi**
Related to the discussion in #2251 

The outcome might just be to leave everything as it is, but if you have any thought please leave them below @ubuntujaggers @Feichtmeier @clobrano @Jupiter007-43 @avil25

My questions on the current icons:
- The Flatpak pictogram is monochrome. Should they all be this way? Or should we add color to the Flatpak icon?
- Is that a Windows logo? Why not change it like Eaglersdeveloper [suggested](https://github.com/snwh/suru-icon-theme/issues/65#issuecomment-399678431)?
- Because the Window files are binary, should they have a solid background too or...?
- Should all of these icons really be packages?
- Or just leave everything as it is 👌 

I have to admit that I'm in a bit over my head here with all the file types, so please be patient 😅 

![image](https://user-images.githubusercontent.com/3986894/88183034-3ac9f800-cc31-11ea-9c27-a088a7166b05.png)

This looks pretty good, but didn't we already add text for zip, rar, 7z? 🤔 
(doh, a package icon got included here) 😄 
![image](https://user-images.githubusercontent.com/3986894/88183082-4e755e80-cc31-11ea-939c-6aec7ccc2809.png)

Images files. Should these have a background color? 
![image](https://user-images.githubusercontent.com/3986894/88183168-6947d300-cc31-11ea-9f8d-bee4bcf34480.png)

There's a few more files we don't have yet which is for VirtualBox. Here are the icons from Suru++
![image](https://user-images.githubusercontent.com/3986894/88185447-3bb05900-cc34-11ea-98e9-adb560552c88.png)

**Comment by clobrano** [comment]
It seems a big work indeed 😀

It seems hard to draw a precise line between mimetypes right? Some are just archive, but others are both archive and packages.

**Comment by Jupi007** [comment]
My opinion:
- It should be great to have color pictogram for Flatpack, to fit with the others.
- The actual W$ logo is very confusing. The Eaglersdeveloper suggestion looks better (maybe add something to show that .exe are executable, otherwise it looks like that file was just... Windows into a file 😈), but do we have rights to put this branding logo here?
- Images files look good, the disk pictogram is very visible and it's fine like that. 👍
- As a Virtualbox user I say yes to add these mimetype icons (reworked for Yaru of course).

I hope this will be useful for you.
Good luck 🍀, if you need help don't hesitated. 🙂

**Comment by Feichtmeier** [comment]
If we keep archives with a zipper and files with text, readable by humans (text files, config files, markdown files, latex files, source code files like c, Java, whatever, markup files like html, XML and so on) like they are now with a piece of paper and a subtle watermark I'm totally fine with changing the others to have more colour.

**Comment by Feichtmeier** [comment]
Sorry i missed the image file question. The disk representing the image might look totally logical for us old dirt bags, but I could imagine that some  younger people do not not what that circle with a dot might be. A female breast? What the hell is that? Etc etc
Anyone has a better idea?
Containers on the other hand could be symbolised by 🚚🚚🚚🚚

**Comment by Jupi007** [comment]
😲 I didn't realize that disk are now like dinosaurs 🦕, and maybe some young people may never seen one... But I haven't got any idea to replace that pictogram (maybe replace it by a fullcolor one should reduce confusion with other things... 😉).

> The disk representing the image might look totally logical for us old dirt bags

I'm still not that old, I'm 18. 😉
But I still lived in the age of CD, DVD, ... and even the floppy disks...

**Comment by ubuntujaggers** [comment]
> If we keep archives with a zipper and files with text, readable by humans (text files, config files, markdown files, latex files, source code files like c, Java, whatever, markup files like html, XML and so on) like they are now with a piece of paper and a subtle watermark I'm totally fine with changing the others to have more colour.

Yeah, I agree that any file that will be opened in a word processor or text editor should have a white background :+1: 

Whereas file extensions that belong to an app can have the brand colours (including the background colour) of the app.

**Comment by madsrh** [comment]
> ...if you need help don't hesitated 🙂

That would be amazing @Jupiter007-43 👼

**Comment by avil25** [comment]
> If we keep archives with a zipper and files with text, readable by humans (text files, config files, markdown files, latex files, source code files like c, Java, whatever, markup files like html, XML and so on) like they are now with a piece of paper and a subtle watermark I'm totally fine with changing the others to have more colour

Yes this makes much more sense, probably I should close #2174 
One question though. What about audio files they also have white background 

For packages, I was thinking about some kind of packed box icon because like in the current deb icon in small size the only thing we can see is a pink color icon.
![iconfinder_Package__Package_Delivery_Box_Parcel_Shipping-02_4072091](https://user-images.githubusercontent.com/40562410/88247445-58dc3a80-ccbb-11ea-869c-34c137d96fcd.png)
I am searching for icon all-day but can not find anything that can represent everything correctly and will still remain recognizable at the smaller size, don't know if something like this can be useful to represent what we want it to.

although disks are now long gone but still img and iso are really tightly coupled with CDs, DVDs don't know if some other icon can represent them as well as a cd icon. It's still used by everyone to represent them and they also have cd/disk in their name.

+1 for adding text below archive and for virtual box files

**Comment by Jupi007** [comment]
And for the disks images what do you think of something like this (just an idea):

![Capture d’écran du 2020-07-23 10-43-09](https://user-images.githubusercontent.com/36476595/88267648-68667e00-ccd1-11ea-9a1d-d6b9174ca22f.png)

**Comment by Feichtmeier** [comment]
def. an improvement over the :white_circle:  with the dot !
if we go with the disk metaphor

**Comment by Jupi007** [comment]
I think that the disk is the only option (except if someone has another idea of course).
If everyone is okay with this design, should I make a PR for all disk image mimetypes?

**Comment by Jupi007** [comment]
And as said @avil25 upper: +1 for adding text below archive, like the good-old Humanity theme:

![Capture d’écran du 2020-07-23 13-16-22](https://user-images.githubusercontent.com/36476595/88280791-c6ea2700-cce6-11ea-85fe-21fccefe2040.png)

🤔 I may be wrong, but wasn't there some text, earlier in Yaru?

**Comment by ubuntujaggers** [comment]
I think we're gravitating towards some principles (correct me if I'm wrong!) and I have some discussion points associated with them.

- Files normally use a portrait backplate:

![image](https://user-images.githubusercontent.com/38893390/89668941-87226280-d8d6-11ea-8d2f-03895fd4c2e9.png) ![image](https://user-images.githubusercontent.com/38893390/89669105-c5b81d00-d8d6-11ea-9bfd-37cb6a7038ea.png) ![image](https://user-images.githubusercontent.com/38893390/89669175-e2eceb80-d8d6-11ea-9d33-739cb84c1321.png)

- ...except for pictures and media, which are sometimes landscape:

![image](https://user-images.githubusercontent.com/38893390/89668180-4544ec80-d8d5-11ea-9da7-3870a7eb011c.png) ![image](https://user-images.githubusercontent.com/38893390/89668257-61e12480-d8d5-11ea-96fd-dba59e57d1d5.png)

**Question 1:** why are these two icons a different ratio at 48px?

**Question 2:** should the Krita and .svg mimetypes and any other portrait-picture icons be changed to landscape?  If not... what's our reason for the difference?

- Files that are primarily opened or viewed as text have a white backplate with subtle watermark (because we want the icon to resemble a page in a document):

![image](https://user-images.githubusercontent.com/38893390/89667942-eed7ae00-d8d4-11ea-92b5-34616b49d2ee.png) ![image](https://user-images.githubusercontent.com/38893390/89667973-0151e780-d8d5-11ea-8a10-f6901a155095.png) ![image](https://user-images.githubusercontent.com/38893390/89668031-1464b780-d8d5-11ea-947d-5bd046a56aad.png)

- Proprietary or branded formats can use the branding, including colour:

![image](https://user-images.githubusercontent.com/38893390/89669187-e6807280-d8d6-11ea-9a69-23d53694f494.png) ![image](https://user-images.githubusercontent.com/38893390/89669202-f13b0780-d8d6-11ea-807d-1978f2faa7a0.png)  ![image](https://user-images.githubusercontent.com/38893390/89669233-0021ba00-d8d7-11ea-8e87-2b1e0cd57514.png)

- Compressed archives have a portrait backdrop with a zip:

![image](https://user-images.githubusercontent.com/38893390/89669447-5e4e9d00-d8d7-11ea-905e-d61d26f6d310.png) ![image](https://user-images.githubusercontent.com/38893390/89669480-70c8d680-d8d7-11ea-9df3-48386249beb1.png) ![image](https://user-images.githubusercontent.com/38893390/89669747-e92f9780-d8d7-11ea-9e7c-f66c6b82bdae.png)

**Question 3:** the Android one is beautiful, but it makes me wonder...  since Rar and Gzip are also branded, should we use their logos or at least colours?  If not: when do we use a logo and when do we use simple text?  We need a policy around the inclusion of file extensions as plain text.

![image](https://user-images.githubusercontent.com/38893390/89670112-8b4f7f80-d8d8-11ea-83aa-406c536601c8.png)
![image](https://user-images.githubusercontent.com/38893390/89670264-d36ea200-d8d8-11ea-8cd0-5442199f414d.png)

- Packages have a portrait backdrop that looks like a parcel:

![image](https://user-images.githubusercontent.com/38893390/89670312-e6817200-d8d8-11ea-85a4-73c08b269be2.png) ![image](https://user-images.githubusercontent.com/38893390/89670350-fb5e0580-d8d8-11ea-8002-afd4d10948a0.png)

**Question 4:** why do snaps and flatpacks not have the tape and barcode behind the logo?

- Lastly, we've made a special case for executables, which have a square backplate.

**Opinion:** IMO, I'm not sure it's wise to use the square baseplate for other mimetypes.  A normal file can be imagined as being a "document" (portrait) or "media" (landscape).  I'm not sure what square means.  I think the executable icon works because it's distinct enough to show the difference between "executables" and mere "files", without looking completely alien to the theme when you see them together in file manager:

![image](https://user-images.githubusercontent.com/38893390/89670904-e5047980-d8d9-11ea-8da3-9b95f2a18584.png)

It basically looks like an icon for an executable rather than a file, which I think is fine (and works in other contexts like the launcher) but should be used very sparingly.  I don't think it's wise to start making lots of mimetype icons that look like apps rather than files?

- As seen throughout the theme, we don't crop or "mask" pictograms.

**Opinion:** I didn't intend for the cogs on the executable icon to be a pictogram, which is why they don't have a drop shadow or highlight.  I see it as a patterned background, like the clouds in the weather icon or the maps in the web browser and map icons.  These are features of the background and are drawn as if they go beyond the icon edges:

![image](https://user-images.githubusercontent.com/38893390/89671827-8f30d100-d8db-11ea-84d8-b40e29381544.png) ![image](https://user-images.githubusercontent.com/38893390/89671894-af609000-d8db-11ea-9c4a-047bb54d878f.png) ![image](https://user-images.githubusercontent.com/38893390/89672418-8391da00-d8dc-11ea-9120-9f45beeb6bda.png) ![image](https://user-images.githubusercontent.com/38893390/89671978-db7c1100-d8db-11ea-9dcc-73d1bab4efec.png)

Again, I think this should be used very sparingly.  If we start adding a lot of icons that look like cropped images, it would change the Yaru style quite a bit?

**Comment by Feichtmeier** [comment]
Question 4:

Snaps and flatpaks are container images, think  of operating system isos, but smaller. So it's okay that they look different from archives

Having zippers for archives is perfect imo, even if they are not compressed. If you pack stuff into a bag with a zipper you also do not always stuff it up until it bursts, but you still need to open the zipper with your hand to get back to the things inside

Gear wars part 2:

I think we should thinkg again about application-x-executable. My problem is not the square baseplate, which makes sense because sadly this icon can be put into the dock, but the gear pictogram reaching until the edges. I think we have no icon, especially no mimetype, in this style (even if it is drawn very professionally)

Question 2: I think this is a good idea, but I would need to see it in a screenshot I guess :)

**Comment by ubuntujaggers** [comment]
> Question 4:
> 
> Snaps and flatpaks are container images, think of operating system isos, but smaller. So it's okay that they look different from archives
> 

:+1:  I don't object to it - just wondered if we had a rationale!

> Having zippers for archives is perfect imo, even if they are not compressed. If you pack stuff into a bag with a zipper you also do not always stuff it up until it bursts, but you still need to open the zipper with your hand to get back to the things inside
> 
> Gear wars part 2:
> 
> I think we should thinkg again about application-x-executable. My problem is not the square baseplate, which makes sense because sadly this icon can be put into the dock, but the gear pictogram reaching until the edges. I think we have no icon, especially no mimetype, in this style (even if it is drawn very professionally)

Always happy to think again on any icon  :)  I don't see it myself, I'm living with the icon on the dock and it looks quite nice.  Is it that different to other elements that reach the edges:

![image](https://user-images.githubusercontent.com/38893390/89681807-91039000-d8ed-11ea-8820-75de95272ff7.png) ![image](https://user-images.githubusercontent.com/38893390/89681975-d7f18580-d8ed-11ea-8bd5-9ff6ee1f3ae4.png)

??

> 
> Question 2: I think this is a good idea, but I would need to see it in a screenshot I guess :)

I'll try it :)

**Comment by ubuntujaggers** [comment]
Oh, another principle:

- Mimetypes should be consistent with the associated app?

**Comment by Feichtmeier** [comment]
Maybe I just need time to get used to it. In my opinion it sticks out compared to the rest of the mimetypes. 
Yes the baseplate of other icons reachs the edge, but for mimetype icons I don't think we have any of these? Anyways, I think it's cool to bring consistency to all the mimetype and maybe even guidelines. Eventually we can come back later to the gear wars ^^

**Comment by ubuntujaggers** [comment]
> Question 2: I think this is a good idea, but I would need to see it in a screenshot I guess :)

A draft PR for testing purposes, with screenshots!

https://github.com/ubuntu/yaru/pull/2317

:+1:

**Comment by madsrh** [comment]
> Question 3: the Android one is beautiful,...

The Android icon is the odd one out. I love the icon, but for consistency (haven't written that word in a while :smile: ) I think we should remove the pictogram and add the text.

 > Gear wars part 2: I think we should think again about application-x-executable. My problem is not the square baseplate, which makes sense because sadly this icon can be put into the dock, but the gear pictogram reaching until the edges. I think we have no icon, especially no mimetype, in this style (even if it is drawn very professionally)

I'll have to disagree. I think it works perfectly with the current version. It is a very special case and I agree with @ubuntujaggers that no other icon should use this baseplate or be drawn in this style. But the gear pictogram reaching until the edges is what makes this not look like a settings app. OFC I'm all ears if there's a better suggestion for the icon design.

If you're writing up the guidelines somewhere, perhaps these should be added:

- Suite application icons can have custom design; like the borders on Adobe PS/AI or Libre Office

- System files (not the right term) should use gray for monochrome pictograms (using the SAME gray, perhaps #888888)
![image](https://user-images.githubusercontent.com/3986894/89705621-bdca9e00-d913-11ea-84b9-19e86a97f48f.png)

- "Grouped icons" (vague term again, sorry) use the same pictogram but adds text for different formats - like sounds, archives, Virtualbox...
This means the new PGP icons should either have the text removed or, like earlier, use the same icon with the text.

**Comment by Jupi007** [comment]
> > Question 3: the Android one is beautiful,...
> 
> The Android icon is the odd one out. I love the icon, but for consistency (haven't written that word in a while smile ) I think we should remove the pictogram and add the text.

+1 👍

For the executable mimetype, as it is a very special icon, I think we could use the crop gears ONLY if we don't reuse this concept on any other icon. This precisely distinguishes it from the other icons.
But as said @madsrh, if someone has another idea.

**Comment by Jupi007** [comment]
Removing the logo from apk archive will looks like this:

![Capture d’écran du 2020-08-08 12-35-00](https://user-images.githubusercontent.com/36476595/89709165-c5bf2800-d97d-11ea-952c-e5c2c90a87b0.png)

PR?

**Comment by clobrano** [comment]
> Oh, another principle:
> 
> - Mimetypes should be consistent with the associated app?

This might be though, because there might be more apps for the same mimetype

**Comment by madsrh** [comment]
> Mimetypes should be consistent with the associated app?
> This might be though, because there might be more apps for the same mimetype

It depends on what this actually means. Perhaps this only applies to app specific files. 😄 What I take away from this line is design the icon with the app in mind. Like "don't use red background for qt files".

**Comment by ubuntujaggers** [comment]
> Removing the logo from apk archive will looks like this:
> 
> ![Capture d’écran du 2020-08-08 12-35-00](https://user-images.githubusercontent.com/36476595/89709165-c5bf2800-d97d-11ea-952c-e5c2c90a87b0.png)
> 
> PR?

Oh no, lol!  The Android one with the robot is one of my favourites!

The alternative (using logos where they exist for other formats) would look something like this:

![image](https://user-images.githubusercontent.com/38893390/89729651-a254b580-da2f-11ea-8e4d-7289dbf9db7d.png)

**Comment by Jupi007** [comment]
> Oh no, lol! The Android one with the robot is one of my favourites!
> 
> The alternative (using logos where they exist for other formats) would look something like this:
> 
> ![image](https://user-images.githubusercontent.com/38893390/89729651-a254b580-da2f-11ea-8e4d-7289dbf9db7d.png)

Yes, I also love the one with the robot. But if we put brand logo on archives, I really don't want to use that thing on an icon 😅:

![gzip logo](https://user-images.githubusercontent.com/38893390/89670264-d36ea200-d8d8-11ea-8cd0-5442199f414d.png)

This is soooo... 😬

Maybe we can make a concession and allow the logo only on the apk because it is like a package (which have brand logo) and keep text on the other archives? 😇

**Comment by ubuntujaggers** [comment]
> Yes, I also love the one with the robot. But if we put brand logo on archives, I really don't want to use that thing on an icon :
> 
> ![gzip logo](https://user-images.githubusercontent.com/38893390/89670264-d36ea200-d8d8-11ea-8cd0-5442199f414d.png)
> 
> This is soooo... 

It is a bit much :rofl: but it's fairly harmless if we use it as a watermark rather than a full coloured logo:

![image](https://user-images.githubusercontent.com/38893390/89730827-2d3aad80-da3a-11ea-8baf-c5cee1b474a0.png)

**EDIT:** Obviously I love the robot :yum: and think branding is cute for proprietary or branded file formats, but I'm happy to go with majority verdict :)

**Comment by ubuntujaggers** [comment]
Something else we could consider as a stretch goal or nice-to-have would be pixel-fitting the mimetype text for greater legibility at 24px and 32px.  It's slightly boring work but does make a tiny of a difference:

![image](https://user-images.githubusercontent.com/38893390/89731761-7f330180-da41-11ea-986b-8ffa89d644b2.png)

Doesn't necessarily look any better but it's easier to see what the text says.

**Comment by Jupi007** [comment]
> It is a bit much rofl but it's fairly harmless if we use it as a watermark rather than a full coloured logo:
> 
> ![image](https://user-images.githubusercontent.com/38893390/89730827-2d3aad80-da3a-11ea-8baf-c5cee1b474a0.png)
> 
> **EDIT:** Obviously I love the robot yum and think branding is cute for proprietary or branded file formats, but I'm happy to go with majority verdict :)

In fact it looks pretty good on your screenshot, so +1 for adding brand logo on archives icons 👍
Just a thought, but IMO the 7zip logo should looks more like the gzip logo which looks like a watermark. 🤔

Otherwise very good job!

**Comment by madsrh** [comment]
Looks great @ubuntujaggers +1 👍 
How does this look in smaller sizes and does all the files have branding?

**Comment by Feichtmeier** [comment]
@madsrh could you eventually put al the useful information out of this issue into the wiki as a new page for mimetype guidelines?

**Comment by Feichtmeier** [comment]
@madsrh is there anything left you want to add to the wiki?

**Comment by madsrh** [comment]
> is there anything left you want to add to the wiki?

I've lost the overview / big picture here, so ATM no, but please @ubuntujaggers or @Jupi007 if you want to add anything here, just go ahead 👍
